packages/components/src/components/select/select/select.tsx

```diff
@@ -0,0 +1,391 @@
+    const selectMessageId = hasMessage(this.host, this.message, this.state) ? messageId : undefined;
+    const ariaDescribedBy =
+      descriptionId && selectMessageId
+        ? `${descriptionId} ${selectMessageId}`
+        : descriptionId || selectMessageId || undefined;
+
+    return (
+      <div class="root">
+        <Label
+          host={this.host}
+          label={this.label}
+          description={this.description}
+          htmlFor={buttonId}
+          isRequired={this.required}
+          isDisabled={this.disabled}
+        />
+        <span class="sr-only">
+          {getSelectedOptionString(this.selectOptions) || 'No option'} selected, {this.selectOptions.length} options in
+          total
+        </span>
+        <div class={{ wrapper: true, disabled: this.disabled }} ref={(el) => (this.comboboxContainer = el)}>
+          <button
+            type="button"
+            role="combobox"
+            id={buttonId}
+            {...getComboboxAriaAttributes(this.isOpen, this.required, labelId, ariaDescribedBy, dropdownId)}
+            disabled={this.disabled}
+            onClick={this.onComboClick}
+            onKeyDown={this.onComboKeyDown}
+            ref={(el) => (this.combobox = el)}
+          >
+            {getSelectedOptionString(this.selectOptions)}
+          </button>
+          <PrefixedTagNames.pIcon
+            class={{ icon: true, 'icon--rotate': this.isOpen }}
+            name="arrow-head-down"
+            theme={this.theme}
+            color={this.disabled ? 'state-disabled' : 'primary'}
+            aria-hidden="true"
+          />
+          <div
+            id={dropdownId}
+            class="listbox"
+            {...getListAriaAttributes(this.label, this.required, false, this.isOpen)}
+            tabindex="-1"
+            ref={(el) => (this.listElement = el)}
+          >
+            <slot />
+          </div>
+        </div>
+        <StateMessage state={this.state} message={this.message} theme={this.theme} host={this.host} />
+        <span class="sr-only" role="status" aria-live="assertive" aria-relevant="additions text">
+          {this.srHighlightedOptionText}
+        </span>
+        {this.isWithinForm && <slot name={INTERNAL_SELECT_SLOT} />}
+      </div>
+    );
+  }
+
+  private onSlotchange = (): void => {
+    this.updateOptions();
+    updateSelectOptions(this.selectOptions, this.value);
+    if (this.isWithinForm) {
+      updateNativeSelectOption(this.nativeSelect, this.selectOptions);
+    }
+    // Necessary to update selected options in placeholder
+    forceUpdate(this.host);
+  };
+
+  private updateOptions = (): void => {
+    this.selectOptions = Array.from(this.host.children).filter(
+      (el) => el.tagName !== 'SELECT' && el.slot !== 'label' && el.slot !== 'description' && el.slot !== 'message'
+    ) as HTMLPMultiSelectOptionElement[];
+    this.selectOptions.forEach((child) => throwIfElementIsNotOfKind(this.host, child, 'p-select-option'));
```

Comment:
Hmm, quite a round trip with n extra cycles when validating like this.  
Instead, prefixed `p-select-option`s could be selected directly and if the remaining children don't match the named slot and `SELECT` condition from above then something is wrong and we should trow.